### PR TITLE
First microwave support

### PR DESCRIPTION
**Pull request by titilambert**
Hello !

This is my first contribution to this cool project !
I'm not sure if I did that correctly :/

I'm trying to add microwave support.
It still misses microwave status (cooking state /cook time left/...) but I'm able to set different settings like: time/sound/weight/light/... and add a new HA service (see screenshots)

image
image

This PR tries to fix https://github.com/ollo69/ha-smartthinq-sensors/issues/385

Please feel free to change my code to make it mode similar to the base code :)

### CODE REVIEW

**Comment by samuelkadolph** [comment]
Following as I recently got an LG ThinQ microwave. I can try to test it out on the weekend.

Also not sure if it's exposed via the API but there's an additional setting for the turntable on/off. You can set it on the microwave itself but not in the ThinQ app so it might not be exposed.

**Comment by samuelkadolph** [comment]
Got it set up and working with my microwave. First observations:

* Turning Clock 24h format on seems to just clear the clock. I see that option is gone from my ThinQ app (it was there last weekend)
* Light has a low and high setting not present in this integration
* I think the weight units should be a dropdown with lbs/kgs instead of a toggle
* App has night light setting, possible to implement?
* App has timer icon in top left for 'oven' timer, I don't think it's worth implementing in HA
* Cooking features, do you want to implement? I see there's microwave/defrost/timed defrost modes, cook time, power strength and 'Send' to start the cook

**Comment by TheAlphaLaw** [comment]
> Got it set up and working with my microwave. First observations:
> 
> * Turning Clock 24h format on seems to just clear the clock. I see that option is gone from my ThinQ app (it was there last weekend)
> * Light has a low and high setting not present in this integration
> * I think the weight units should be a dropdown with lbs/kgs instead of a toggle
> * App has night light setting, possible to implement?
> * App has timer icon in top left for 'oven' timer, I don't think it's worth implementing in HA
> * Cooking features, do you want to implement? I see there's microwave/defrost/timed defrost modes, cook time, power strength and 'Send' to start the cook

Just FYI on cook time, it can only add time to an already cooking microwave but not start a new cook.

**Comment by samuelkadolph** [comment]
> Just FYI on cook time, it can only add time to an already cooking microwave but not start a new cook.

That's not what I'm seeing with the ThinQ. You pick the settings you want and press 'Send'. It sends them over to the microwave. Then you have to press start. When it's running, the app only shows the status and you can't change it.

**Review by ollo69 — CHANGES_REQUESTED** [review]
Thank you very match for this PR. Please take a look to my review comments.

**Comment by samuelkadolph** [comment]
24 hour time select still just turns the clock (if off) and sets it to 12:00 AM. The setting is gone from the ThinQ app so I don't think it's a thing anymore? I installed my new ThinQ range this weekend and it does have a (working) 24 hour clock setting. Annoying they aren't consistent.

File: custom_components/smartthinq_sensors/wideq/factory.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/wideq/factory.py]: fixed

File: custom_components/smartthinq_sensors/wideq/factory.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/wideq/factory.py]: fixed

File: custom_components/smartthinq_sensors/wideq/const.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/wideq/const.py]: fixed

File: custom_components/smartthinq_sensors/__init__.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/__init__.py]: fixed

File: custom_components/smartthinq_sensors/__init__.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/__init__.py]: fixed

**Comment by ollo69** [comment]
@titilambert, any news on changes requested?

**Comment by titilambert** [comment]
> @titilambert, any news on changes requested?

I fixed the easy ones, Trying now to improve the set_time services as suggested.
I don't understand which wash service should I "copy"

File: custom_components/smartthinq_sensors/services.yaml
Comment (titilambert) [reply custom_components/smartthinq_sensors/services.yaml]: fixed

File: custom_components/smartthinq_sensors/__init__.py
Comment (titilambert) [reply custom_components/smartthinq_sensors/__init__.py]: fixed

**Comment by titilambert** [comment]
@ollo69 I finally got the service done as suggested.
I will try to fix the last two comment in few the days

**Comment by samuelkadolph** [comment]
Updating to the latest (8012b34) removes my microwave from HASS. I have the MVEL2033F. Shows up as WMVEL2137 from the LG API. I'll upload my diagnostics file for the microwave.

[smartthinq_sensors-33e935dee672f618b22a784d3e770f66-Microwave-a0ef3857fc07022ae5c09b1957ec8ba4.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/11931817/smartthinq_sensors-33e935dee672f618b22a784d3e770f66-Microwave-a0ef3857fc07022ae5c09b1957ec8ba4.json.txt)

File: custom_components/smartthinq_sensors/device_helpers.py
Comment (ollo69) [inline custom_components/smartthinq_sensors/device_helpers.py:311]: This is useless, there are no dedicated method. Just use `LGEBaseDevice`.

File: custom_components/smartthinq_sensors/__init__.py
Comment (ollo69) [inline custom_components/smartthinq_sensors/__init__.py:80]: I think this is not used anymore

**Review by ollo69 — CHANGES_REQUESTED** [review]

File: custom_components/smartthinq_sensors/sensor.py
Comment (ollo69) [inline custom_components/smartthinq_sensors/sensor.py:863]: I think also this is useless, `LGESensor` should already implement this logic for `DEFAULT_SENSOR`

**Review by ollo69 — CHANGES_REQUESTED** [review]

**Comment by ollo69** [comment]
Please take a look to failing `Linting` test.

**Comment by ollo69** [comment]
@titilambert,

when do you think you will be able to complete this PR? I would make some changes to codebase to respect last HA changes but this will create conflict with PR, so should be better to merge before.

**Comment by titilambert** [comment]
> @titilambert,
> 
> when do you think you will be able to complete this PR? I would make some changes to codebase to respect last HA changes but this will create conflict with PR, so should be better to merge before.

Hello, I do not know when I will be able to achieve it I don't have a lot of time, and I'm stuck with Fan state :/

Can you merge it then disable the microwave ?
I can continue after your changes.

Thanks !

**Comment by titilambert** [comment]
> @titilambert,
> 
> when do you think you will be able to complete this PR? I would make some changes to codebase to respect last HA changes but this will create conflict with PR, so should be better to merge before.

Hello @ollo69, I do not know when I will be able to achieve it ... I don't have a lot of time, and I'm stuck with Fan status :/

Can you merge it then disable the microwave ?
I can continue after your changes.

Thanks !

**Comment by trentjw** [comment]
any progress on this I have a over the range microwave that is waiting to find its way into home assistant thanks

**Comment by teek541** [comment]
I yoo am interested in this.  Would ve nice to set timers through HA and control automations with it.

**Comment by trentjw** [comment]
well the model number of my microwave is MHEC1737D if that makes any difference though this seems to have lost momentum

**Comment by teek541** [comment]
I have the same one!  Just a different color. lol

**Comment by trentjw** [comment]
you got the MHEC1737F one eh?

**Comment by ollo69** [comment]
All, 

I'm currently working on this PR, but it require times. Please be patient.

**Comment by teek541** [comment]
Thanks for the update, I'll do a test.  My network switch just bricked itself last night so I'm waiting on my replacement first.

**Comment by trentjw** [comment]
> Thanks for the update, I'll do a test. My network switch just bricked itself last night so I'm waiting on my replacement first.

its not uploaded yet so if it is then it does not work

**Comment by ollo69** [comment]
> its not uploaded yet so if it is then it does not work

What do you mean? This PR is still not merged, to test this you must update your installation with this branch.

**Comment by TheAlphaLaw** [comment]
> > Thanks for the update, I'll do a test. My network switch just bricked itself last night so I'm waiting on my replacement first.
> 
> its not uploaded yet so if it is then it does not work

@trentjw Sounds like you may not be familiar with Github or how Home Assistant operates as an open source platform. If you're not pulling code to test manually, then I advise that you be patient and wait for the PR to be merged and Home Assitant updated. ollo69 is doing this in his free time in an effort to help the community.

**Comment by trentjw** [comment]
> > > Thanks for the update, I'll do a test. My network switch just bricked itself last night so I'm waiting on my replacement first.
> > 
> > 
> > its not uploaded yet so if it is then it does not work
> 
> @trentjw Sounds like you may not be familiar with Github or how Home Assistant operates as an open source platform. If you're not pulling code to test manually, then I advise that you be patient and wait for the PR to be merged and Home Assitant updated. ollo69 is doing this in his free time in an effort to help the community.

always one in the crowd that has to inject themselves and offer nothing

**Comment by teek541** [comment]
> > > Thanks for the update, I'll do a test. My network switch just bricked itself last night so I'm waiting on my replacement first.
> > 
> > 
> > its not uploaded yet so if it is then it does not work
> 
> @trentjw Sounds like you may not be familiar with Github or how Home Assistant operates as an open source platform. If you're not pulling code to test manually, then I advise that you be patient and wait for the PR to be merged and Home Assitant updated. ollo69 is doing this in his free time in an effort to help the community.

Might I suggest you tell us how to do that as I too am new and assumed that this would have shown up as an update for me to install when I got home and got a new working switch installed? :-D  Do I just use this as a new custom repository?

**Comment by zimmra** [comment]
@ollo69 

In attempt to get this discussion back on track to this PR...

Here are my findings with the current PR code

I deleted the SmartThinq LGE Sensors integration from Home Asssistant
I shutdown Home Assistant then deleted my existing smartthinq_sensors folder and then fetched/checked out this PR
I then started my Home Assistant instance, and then re-added the integration as a new integration
I'm not home to test the sensor states, but am able to test everything else

## Fan Entity

- Vent entity shows up, but looks like it is unavailable. 
	- Changing the Vent Speed in the app has no effect on this
	- image
- Light entity does not show up at all

## Configuration Switch/Select Entities

- These all populate and do set the specific config setting on the microwave**
- **Only the config entity you change is set, then everything else goes to a default value
	- if anything other than `Clock Display` is changed, `Clock Display` changes to `On`/`Show`
	- If anything other than `Display Scroll Speed` is changed, `Display Scroll Speed` changes to `Slow`
	- If anything other than `Sound` is changed, `Sound` is changed to `Off`
	- If anything other than `Weight Unit` is changed, `Weight Unit` changes to `KG`
- For example, if I change `Weight Unit` to `LB` the setting is reflected in the app correctly, but then the 3 other applicable config entities get changed to the default value listed above
- image

## Sensor Entites
- I am unable to test the sensor entities below until I am home with the unit
	- sensor.microwave, sensor.microwave_oven_mode, sensor.microwave_oven_state

**Comment by ollo69** [comment]
> Might I suggest you tell us how to do that as I too am new and assumed that this would have shown up as an update for me to install when I got home and got a new working switch installed? :-D Do I just use this as a new custom repository?

@teek541,

until this PR is not merged, the only way to test this is to manually install integration taking the code from this branch. This is not so simple as install from HACS and if you don't have experience it will be simple to make mistake. I you don want to take risk, just wait for tests finish and new release available.

**Comment by ollo69** [comment]
> On an Over-The-Range Microwave, you use the vent when the range (stove) below is in use

So the vent can be always turned on or there are some condition when it is not possible?

> I could be mistaken, but I thought I read on the LG api documentation that it is expecting all of them to be sent

I will try to change implementation to always send all values, anyway doesn't seems a good approach always set time. It would be good if you will be able to do a mitm

**Comment by ollo69** [comment]
Ok, I implemented some changes based on test results, please try again and give me your feedback

**Comment by zimmra** [comment]
I got mitm proxy setup today
**It appears settings are all set even when individual setting is changed**

Here's what I've poked in to so far, I think this should be conclusive enough - but let me know if you want any other information @ollo69 

I will also try out your changes

## Turned vent on (direct to Low) in app

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 1
        }
    },
    "dataValue": null
}
```

## Turned Light On (direct to low) in App

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 1,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "DISABLE",
            "mwoVentSpeedLevel": 0
        }
    },
    "dataValue": null
}
```

## Turned clock from Show -> Hide in app

```json
{
    "command": "Set",
    "ctrlKey": "SetPreference",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_SETTING",
            "cmdOptionDataLength": "REMOTE_SETTING",
            "mwoSettingClockDisplay": "CLOCK_HIDE",
            "mwoSettingClockSetHourMode": "NOT_SET",
            "mwoSettingClockSetTimeHour": 128,
            "mwoSettingClockSetTimeMin": 128,
            "mwoSettingClockSetTimeSec": 128,
            "mwoSettingDefrostWeightMode": "NOT_SET",
            "mwoSettingDemoMode": "NOT_SET",
            "mwoSettingDisplayScrollSpeed": "NOT_SET",
            "mwoSettingSound": "NOT_SET"
        }
    },
    "dataValue": null
}
```

## Turned clock from Hide -> Show in app

```json
{
    "command": "Set",
    "ctrlKey": "SetPreference",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_SETTING",
            "cmdOptionDataLength": "REMOTE_SETTING",
            "mwoSettingClockDisplay": "CLOCK_SHOW",
            "mwoSettingClockSetHourMode": "NOT_SET",
            "mwoSettingClockSetTimeHour": 128,
            "mwoSettingClockSetTimeMin": 128,
            "mwoSettingClockSetTimeSec": 128,
            "mwoSettingDefrostWeightMode": "NOT_SET",
            "mwoSettingDemoMode": "NOT_SET",
            "mwoSettingDisplayScrollSpeed": "NOT_SET",
            "mwoSettingSound": "NOT_SET"
        }
    },
    "dataValue": null
}
```

## Set clock time in app

```json
{
    "command": "Set",
    "ctrlKey": "SetPreference",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_SETTING",
            "cmdOptionDataLength": "REMOTE_SETTING",
            "mwoSettingClockDisplay": "NOT_SET",
            "mwoSettingClockSetHourMode": "NOT_SET",
            "mwoSettingClockSetTimeHour": 15,
            "mwoSettingClockSetTimeMin": 55,
            "mwoSettingClockSetTimeSec": 24,
            "mwoSettingDefrostWeightMode": "NOT_SET",
            "mwoSettingDemoMode": "NOT_SET",
            "mwoSettingDisplayScrollSpeed": "NOT_SET",
            "mwoSettingSound": "NOT_SET"
        }
    },
    "dataValue": null
}
```

**Comment by ollo69** [comment]
> I got mitm proxy setup today

Great, this is really useful.

I suggest to wait that I align the code with discovered commands before trying my last changes, now is clear to me how to properly manage preference.

Thanks a lot.

**Comment by zimmra** [comment]
> > I got mitm proxy setup today
> 
> Great, this is really useful.
> 
> I suggest to wait that I align the code with discovered commands before trying my last changes, now is clear to me how to properly manage preference.
> 
> Thanks a lot.

Glad to help!

I just updated my comment to add a few more entries (most notably `Timer`, but also cook time and defrost info), as well as made it collapsible to make the discussion page here a bit more managable. 

Please let me know if there's anything I can pull from it for you, though I think I covered the main control functions.  

Thanks again for the hard work!

**Comment by ollo69** [comment]
@zimmra,

could you please make a last test? What is the command sent if you turn on the vent when the light is on (or viceversa the command sent to turn on the lamp when the vent is on).
Not sure if `DISABLE` means "turn off" or "ignore the command".

Thanks.

**Comment by zimmra** [comment]
## Light Turned On (VENT ALREADY ON)

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 1,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 1
        }
    },
    "dataValue": null
}
```

## Vent Turned On (LIGHT ALREADY ON)

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 1,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 1
        }
    },
    "dataValue": null
}
```

## Vent turned off, light left on

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 1,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "DISABLE",
            "mwoVentSpeedLevel": 0
        }
    },
    "dataValue": null
}
```

## Light turned off, vent left on

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 1
        }
    },
    "dataValue": null
}
```

**Comment by ollo69** [comment]
Ok, now it should be ready for merge, just wait for final test.
I also restored the `set_time` service because now are clear required parameters.

**Comment by zimmra** [comment]
> just wait for final test.

### Light (select) entity turns light on/off, but always goes to low

- WHEN SET TO HIGH IN LG APP - Home Assistant updates the select entity state **correctly**
- WHEN SET TO HIGH IN HOME ASSISTANT - Home Assistant initially shows the state as the selected state, but it is actually set to `LOW` in reality/the LG app. This eventually updates to `LOW` in Home Assistant

### Vent Speed (select) entity turns vent on/off, but always goes to high 

- WHEN SET TO ANYTHING OTHER THAN HIGH IN LG APP - Home Assistant updates the select entity state **correctly**
- WHEN SET TO ANYTHING OTHER THAN HIGH IN HOME ASSISTANT - Home Assistant initially shows the state as the selected state, but it is actually set to `HIGH` in reality/the LG app. This eventually updates to `HIGH` in Home Assistant

<details>
**<summary>Additional Control Commands (Expandable)</summary>**

All control commands sent to `https://aic-service.lgthinq.com:46030/v1/service/devices/{redacted}/control-sync`

## Fan to Low 

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 1
        }
    },
    "dataValue": null
}
```

## Fan to Mid

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 2
        }
    },
    "dataValue": null
}
```

## Fan to High

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 3
        }
    },
    "dataValue": null
}
```

## Fan to Turbo

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 0,
            "mwoLampOnOff": "DISABLE",
            "mwoVentOnOff": "ENABLE",
            "mwoVentSpeedLevel": 4
        }
    },
    "dataValue": null
}
```

## Light to Low

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 1,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "DISABLE",
            "mwoVentSpeedLevel": 0
        }
    },
    "dataValue": null
}
```

## Light to High

```json
{
    "command": "Set",
    "ctrlKey": "setVentLampLevel",
    "dataGetList": null,
    "dataKey": null,
    "dataSetList": {
        "ovenState": {
            "cmdOptionContentsType": "REMOTE_VENT_LAMP",
            "cmdOptionDataLength": "REMOTE_VENT_LAMP",
            "mwoLampLevel": 2,
            "mwoLampOnOff": "ENABLE",
            "mwoVentOnOff": "DISABLE",
            "mwoVentSpeedLevel": 0
        }
    },
    "dataValue": null
}
```

</details>

**Comment by ollo69** [comment]
> Light (select) entity turns light on/off, but always goes to low

Probably level must be provided as `int` and not as `string`. Please try again.

Thanks

**Comment by ollo69** [comment]
> If you ever need other people to get info from a mitm setup, I logged what I did for my own future reference [here](https://github.com/zimmra/frida-rootbypass-and-sslunpinning-lg-thinq), feel free to point people to it if you need API info in the future. All they need is a WSA-capable Windows 11 PC, and know how to use terminal

This is a great info and would probably help a lot for future implementation in this integration. If is ok for you I would add this information in read-me (or if you prefer you can create a PR for this).

Thanks again.

**Comment by zimmra** [comment]
>If is ok for you I would add this information in read-me (or if you prefer you can create a PR for this).

Like I said, glad to help! Feel free to use/modify/scrap https://github.com/ollo69/ha-smartthinq-sensors/pull/582 

>Probably level must be provided as int and not as string. Please try again.

That seems to have done it, all levels are properly controllable.

Everything _seems_ to be functioning as intended in this iteration

**Comment by trentjw** [comment]
Tried this out the light control works and is controllable but the Fan shows unavailable and does not work

**Comment by ollo69** [comment]
@trentjw ,

this PR is closed and merged and new integration release is available, so please do not post here but open a dedicated issue providing detail and attaching integration diagnostic.

Thanks👍

**Pull request merged**
